### ISSUE TITLE
Add a border to the menu list component

### ISSUE THREAD

**Issue by craigcook** [mozilla/protocol#550]
## Please describe the use case(s) for this component

The [menu list component](https://protocol.mozilla.org/patterns/molecules/menu-list.html) presents a dropdown list of links. Though it looks like a form `select` element, the nuanced difference is that a `select` submits an option through a form handler and a menu list just holds direct links to a URLs.

### Please describe the change you wish to see

Form selects are being designed in #435 and look a lot like the current styling for menu lists, but with the addition of a border. We should add an optional border for the menu list component as well, so we can match it to the visual style of a form select even though it functions differently under the hood.

I'd recommend making it a theme option with an extension class, something like `mzp-t-border` or a more meaningful name if we can think of one. Or perhaps the border is the default and we can make the borderless style the variant.

### Please describe why these changes need to be made

It will be consistent with the styling of form selects and can make menu lists stand out more on a page than the current borderless version. 

I propose making the border optional because sometimes we want it to stand out when it's a primary action, but sometimes we may want it to be more subtle when it's a secondary action or collected with other links.

### Please provide any designs or prototypes of the proposed changes

https://user-images.githubusercontent.com/3740746/67815209-8b764100-fa63-11e9-91fc-f5fc78874f0c.png

### Please describe where this component is currently in use

https://www.mozilla.org/en-US/firefox/
https://www.mozilla.org/en-US/firefox/browsers/compare/chrome/

### Is the development of this component a blocking dependency for other work? Please explain if so

Not necessarily a blocker but the need came about from [design work on the Firefox Enterprise page](https://github.com/mozilla/bedrock/issues/8577). That mockup is using the form style but functionally it's a list of links. We could use the menu list on that page without the border but it likely would get lost among all the other links. It would be good to have this update landed in Protocol quickly for use on the Enterprise page.

**Comment by craigcook** [comment]
There's some ongoing discussion on whether this component should match the form select or look different. An argument in favor of matching is that users only care that "clicking this thing opens a dropdown of options" and don't necessarily distinguish a form dropdown from a menu. The counterargument is that that is a *bad* thing, and that they *should* be able to tell the distinguish them because the two things behave differently (one is navigation, one is part of a form).

Personally, I'm starting to swing to the "they should look different" side because I'm reminded of people using form `selects` for navigation as a short-cut to a dropdown nav, which I think isn't right. If you wouldn't use a form as navigation, you shouldn't use navigation in a form.

I do think we need an option to make the menu list "pop" more and stand out on a page, but I think it should be somehow distinct from a form select. Maybe that means changing our form select styling so it doesn't look so much like a button?

We're going to do a little research and see what the de-facto standard is, if any.

**Comment by Aakash2408** [comment]
I have made a pull request( #583). This pull request contains this issue as well as #582 .Kindly review :)

**Comment by stephaniehobson** [comment]
Agreed on adding a border. Need to backport the design on /firefox and /enterprise.
